src/pykeen/training/training_loop.py

```diff
@@ -118,6 +125,16 @@ def device(self):  # noqa: D401
         """The device used by the model."""
         return self.model.device
 
+    @property
+    def checksum(self) -> int:  # noqa: D401
+        """The checksum of the model and optimizer the training loop was configured with."""
+        # Create a checksum that will give the same results on every system and python version.
+        # https://docs.python.org/3/library/zlib.html
+        return (
+            adler32(str(self.model).encode('utf-8')) & 0xffffffff
+            + adler32(str(self.optimizer).encode('utf-8')) & 0xffffffff
```

Comment:
I am not sure why we need the bit-wise and. Could you add a comment here?

Also, does `str(model)` capture everything relevant?

Reply:
This is based on the note here https://docs.python.org/3/library/zlib.html#zlib.adler32.
I consider this a quick test, avoiding that some obvious mistakes are made. It returns the used model, loss, regularizer as well as the embedding sizes.
However, the model, loss and regularizer are hashed without their parameters, so this could be added if considered necessary.

Reply:
Also, why not use something from `hashlib` like the md5 hash? The following code would give a md5 hash string

```python
import hashlib
h = hashlib.md5()
h.update(str(self.model).encode('utf-8'))
h.update(str(self.optimizer).encode('utf-8'))
return h.hexdigest()
```

Reply:
Doesn't really make a difference as long it delivers the same hash each time. But I'm supporting this suggestions, since it allows us to get rid off the bit-magic of zlib.adler32